java/client/src/org/openqa/selenium/support/ui/Select.java

```diff
@@ -245,11 +237,16 @@ public void deselectAll() {
   public void deselectByValue(String value) {
     List<WebElement> options = element.findElements(By.xpath(
         ".//option[@value = " + Quotes.escape(value) + "]"));
-
+    boolean matched = false;
     for (WebElement option : options) {
-      if (option.isSelected()) {
-        option.click();
+      setSelected(option, false);
+      if (!isMultiple()) {
```

Comment:
deselecting a non-multiple select doesn't make sense. What would it even mean to deselect something in a normal select? A user can't do it.

Rather than adding this check here (same goes for deselectByVisibleText), we could check at the start of the method and throw an exception if it is not multiple.